### PR TITLE
Upgrade version of mypy

### PR DESCRIPTION
**Pull request by zzz4zzz**
- Upgrade version of mypy
- Fix typing errors

### CODE REVIEW

File: eli5/keras/gradcam.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def _validate_target(target, output_shape):
         output_nodes = output_shape[1:][0]
         if not (0 <= target < output_nodes):
             raise ValueError('Prediction target index is ' 
-                             'outside the required range [0, {}). ',
+                             'outside the required range [0, {}). '
```
Comment (lopuhin) [inline eli5/keras/gradcam.py]: Thanks for the catch 👍

File: eli5/formatters/utils.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def tabulate(data,  # type: List[List[Any]]
 
 def format_weight(value):
     # type: (Real) -> str
-    return '{:+.3f}'.format(value)
+    return '{:+.3f}'.format(value)  # type: ignore
```
Comment (lopuhin) [inline eli5/formatters/utils.py]: I see that this is due to https://github.com/python/mypy/issues/3186, maybe a better fix would be to use ``Union[int, float]`` here and below instead of the ``Real`` until that issue is fixed, especially since we use ``Real`` only in this module

File: eli5/formatters/text.py [hunk 3]
```diff
@@ -159,7 +159,7 @@ def _transition_features_lines(explanation):
     return [
         "",
         "Transition features:",
-        tabulate(tf.coef, headers=tf.class_names, showindex=tf.class_names,
+        tabulate(tf.coef, headers=tf.class_names, showindex=tf.class_names,  # type: ignore
```
Comment (lopuhin) [inline eli5/formatters/text.py]: Looks like mypy is confused by local ``tabulate`` import shadowing the global import, probably a better fix would be to move ``from tabulate import tabulate`` to the top level and import ``tabulate`` from utils as ``eli5_tabulate`` or something like this?

File: eli5/formatters/utils.py (hunk 2 above)
Comment (zzz4zzz) [reply eli5/formatters/utils.py]: I changed `Real` to `float`. No need to use `Union[float, int]` ( https://www.python.org/dev/peps/pep-0484/#the-numeric-tower ).

File: eli5/formatters/text.py (hunk 3 above)
Comment (zzz4zzz) [reply eli5/formatters/text.py]: Yes, good point. Changed.

**Comment by zzz4zzz** [comment]
@lopuhin I've checked your comments and made fixes. Additionaly I've added [--ignore-missing-imports](https://mypy.readthedocs.io/en/latest/command_line.html#cmdoption-mypy-ignore-missing-imports) to mypy command line and cleaned up imports.

**Comment by lopuhin** [comment]
Thanks @zzz4zzz , and ``--ignore-missing-imports`` is also a welcome addition. I'll check PR in more detail later today or tomorrow.

**Comment by lopuhin** [comment]
Thanks for review @Gallaecio 👍
